### PR TITLE
fix super group bug and add network proxy

### PR DESCRIPTION
**Pull request by lizqwerscott**
A make-chat error occurs when a bot joins a supergroup.

I want to use a proxy to access Telegram Api, so I added the proxy option.

### CODE REVIEW

File: src/network.lisp [hunk 1]
```diff
@@ -34,16 +41,21 @@
                                       and
                                         collect value))
            (response
-             (dexador:post url
+             (if *proxyp*
```
Comment (svetlyak40wt) [inline src/network.lisp:44]: You don't need addional flag, just check if proxy not nil.

**Review by svetlyak40wt — COMMENTED** [review]
I'd make a few fixes

File: src/network.lisp [hunk 2]
```diff
@@ -7,9 +7,16 @@
                 #:get-endpoint)
   (:export
    #:make-request
-   #:request-error))
+   #:request-error
+   #:set-proxy))
 (in-package cl-telegram-bot/network)
 
+(defparameter *proxyp* nil)
```
[the remark this answers is not in the document]
Comment (lizqwerscott) [reply src/network.lisp:14]: Thanks for the modification, I'm a newbie with Common Lisp.

File: src/network.lisp (hunk 1 above)
Comment (lizqwerscott) [reply src/network.lisp:44]: Thanks for the modification, I'm a newbie with Common Lisp.

**Comment by svetlyak40wt** [comment]
Thank you for the PR!

**Pull request merged**